modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/odata/ExpressionVisitorImpl.java

```diff
@@ -348,6 +384,46 @@ private String _getPropertyName(Object object) {
 		return String.valueOf(object);
 	}
 
+	private Object _sub(Object left, Object right)
+		throws ExpressionVisitException {
+
+		if ((left instanceof String) && (right instanceof Duration)) {
+			try {
+				Date date = null;
+
+				if (Objects.equals(
+						MethodExpression.Type.NOW.toString(), left)) {
+
+					date = new Date();
```

Comment:
As you calculate the date in this moment, he result of a expression that has two nows, will have diferrents datetimes.

And I don't think this is the expected result.

Reply:
Yes, you are right. I haven't thought in that use case,   well seen. 

Thanks

Reply:
Done